OptimizelySDK.Tests/OdpTests/LruCacheTest.cs

```diff
@@ -0,0 +1,201 @@
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // Lookup should move user1 to bottom of the list and push up others.
+            Assert.AreEqual("user2", itemKeys[0]);
+            Assert.AreEqual("user3", itemKeys[1]);
+            Assert.AreEqual("user1", itemKeys[2]);
+
+            Assert.AreEqual(_segments3And4, cache.Lookup("user2"));
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // Lookup should move user2 to bottom of the list and push up others.
+            Assert.AreEqual("user3", itemKeys[0]);
+            Assert.AreEqual("user1", itemKeys[1]);
+            Assert.AreEqual("user2", itemKeys[2]);
+
+            Assert.AreEqual(_segments5And6, cache.Lookup("user3"));
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // Lookup should move user3 to bottom of the list and push up others.
+            Assert.AreEqual("user1", itemKeys[0]);
+            Assert.AreEqual("user2", itemKeys[1]);
+            Assert.AreEqual("user3", itemKeys[2]);
+        }
+
+        [Test]
+        public void ShouldReorderListOnSave()
+        {
+            var cache = new LruCache<List<string>>();
+            
+            cache.Save("user1", _segments1And2);
+            cache.Save("user2", _segments3And4);
+            cache.Save("user3", _segments5And6);
+
+            var itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            Assert.AreEqual("user1", itemKeys[0]);
+            Assert.AreEqual("user2", itemKeys[1]);
+            Assert.AreEqual("user3", itemKeys[2]);
+
+            cache.Save("user1", _segments1And2);
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // save should move user1 to bottom of the list and push up others.
+            Assert.AreEqual("user2", itemKeys[0]);
+            Assert.AreEqual("user3", itemKeys[1]);
+            Assert.AreEqual("user1", itemKeys[2]);
+
+            cache.Save("user2", _segments3And4);
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // save should move user2 to bottom of the list and push up others.
+            Assert.AreEqual("user3", itemKeys[0]);
+            Assert.AreEqual("user1", itemKeys[1]);
+            Assert.AreEqual("user2", itemKeys[2]);
+
+            cache.Save("user3", _segments5And6);
+
+            itemKeys = cache.ReadCurrentCache().Keys.ToKeyCollection();
+            // save should move user3 to bottom of the list and push up others.
+            Assert.AreEqual("user1", itemKeys[0]);
+            Assert.AreEqual("user2", itemKeys[1]);
+            Assert.AreEqual("user3", itemKeys[2]);
+        }
+
+        [Test]
+        public void ShouldHandleWhenCacheIsDisabled()
+        {
+            var cache = new LruCache<List<string>>(0, LruCache<object>.DEFAULT_TIMEOUT_SECONDS);
+
+            cache.Save("user1", _segments1And2);
```

Comment:
Is this possible if we can return `true` or `false`. @jaeopt a question for you?